resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/DriverServiceFeatureStep.scala

```diff
@@ -50,6 +50,8 @@ private[spark] class DriverServiceFeatureStep(
       s"$shorterServiceName as the driver service's name.")
     shorterServiceName
   }
+  // expose driver service name
+  kubernetesConf.sparkConf.set(RESOLVED_SPARK_DRIVER_SVC_NAME, resolvedServiceName)
```

Comment:
Thank you for update, @zwangsheng .

BTW, this proposal looks exceptional because we have been using `kubernetesConf.sparkConf` as read-only until now. Is there any other instance where Apache Spark use `sparkConf` like this?

Reply:
I agree with you that Apache Spark haven't tried to break this read-only rule elsewhere for `kubernetesConf.sparkConf`, so maybe we can try adding an additional variable to kubernetesConf for contextual passing.

I will make changes to the logic inside the PR to comply with the `kubernetesConf.sparkConf`  read-only principle.